### DISCUSSION TITLE
Media Collection

### DISCUSSION THREAD

**Discussion by Deekerman** [tteck/Proxmox#292] [Q&A]
Is there currently any plans to do any or all of the *arrs?
Radarr
Sonarr 
ect.

or Sabnzbd or getnzb? 

I already have these running but i have a few friends I am trying to get into the server game and LOVE proxmox and your scripts so that would make it a lot easier for them. I can provide a completed list of all the programs if that helps or is even considered. If not Thank you for everything you have and continue to do.

**Comment by Deekerman** [comment]
Thanks for the move. I am not great at github yet....or any of this tbh, i stumble my way though.

**Comment by schiz0phr3ne** [comment]
You can use the Debian install script from @tteck and then run *arr community script : https://wiki.servarr.com/install-script

**Reply by Deekerman** [reply]
Does that allow you to install only one or do you have to do them all?

**Reply by schiz0phr3ne** [reply]
It's one by one :)

**Reply by schiz0phr3ne** [reply]
For Sonarr, just need to follow instructions here : https://sonarr.tv/#downloads-v3-linux-debian

**Comment by nacho-domatix** [comment]
https://zerodya.net/self-host-jellyfin-media-streaming-stack/ Amazing

**Comment by drthanwho** [comment]
The question I would have with having all the arr's separated on their own LXCs would be how do you have shared storage? I have the same issue with e.g. nextcloud and plex (wanting to play through plex whatever nextcloud syncs). But I don't want to make them privileged so with unprivileged comes the issue of different permissions for each LXC. I can't for the life of me workout a good solution. How do you guys do storage of media for the arr's?

**Reply by nigiriemoji** [reply]
> The question I would have with having all the arr's separated on their own LXCs would be how do you have shared storage? I have the same issue with e.g. nextcloud and plex (wanting to play through plex whatever nextcloud syncs). But I don't want to make them privileged so with unprivileged comes the issue of different permissions for each LXC. I can't for the life of me workout a good solution. How do you guys do storage of media for the arr's?

I have mount points within an XXX.conf file. My drives are named 14TB and 8TB

```
## Jellyfin LXC
#  ### https%3A//tteck.github.io/Proxmox/
#  <a href='https%3A//ko-fi.com/D1D7EP4GF'><img src='https%3A//img.shields.io/badge/%E2%98%95-Buy me a coffee-red' /></a>
arch: amd64
cores: 4
features: nesting=1
hostname: jellyfin
memory: 2048
mp0: /mnt/14TB,mp=/mnt/14TB
mp1: /mnt/8TB,mp=/mnt/8TB
```

**Reply by drthanwho** [reply]
> I have mount points within an XXX.conf file. My drives are named 14TB and 8TB
> 
> ```
> ## Jellyfin LXC
> ```

Right but that's for Jellyfin. That one only needs to read data, which technically is true for Plex as well actually. But I found that Plex wasn't able to see any files synced by nextcloud. 

What do you do about the other arrs? Do you run them privileged?

**Reply by oOStroudyOo** [reply]
They would have just used Jellyfin as an example, I have the same set up as yslupdates.
Mounted drives on all of the Arrs containers and the Plex container.

**Reply by drthanwho** [reply]
Privileged LXCs or not though? No one is mentioning that 😅

**Reply by oOStroudyOo** [reply]
I use privileged.
But I don't believe you need to.

I believe the method that yslupdates metioned works for either.

Most of my containers are currently mounted by FSTAB which was my old setup but will eventually move them to being mounted via the lxc.conf files as I've mounted the drives to the server now.

**Reply by drthanwho** [reply]
I noticed Jellyfin is isntallled as priviliged so it doesn't have the limitation of needing write access. I also installed the arrs and noticed they all have the same UID/GID inside the container (65534) so can easily have the same mountpoint and give one folder the same permissions to work for all of them. 

That being said the isntructions on how to correctly do remapping are still confusing to me. I've only made it to work by `chown` the directory in the host from root 0 to the equivelent for the LXCs (65534 inside the LXC is 165534 in the host).

My questions were stemming from my attempts at nextcloud and plex sharing a directory for videos/photos to show on the TV through Plex. Those two LXCs didn't share the same UID/GID, which was the issue. In the end I installed them inside a docker LXC (although would prefer to get it to work as two seperate LXCs)

**Reply by oOStroudyOo** [reply]
I understand your preference.
I use to have all of the Arrs in 1 container but then that meant a broken container broke multiple apps.

I can't say that I've heard of nextcloud before to be honest so I'm not sure what it does or why you need it?

**Reply by lordalisat** [reply]
For unpriviliged containers you can use idmapping in the config. I have set up my *arr lxcs to all use the `media` group (1000), and added the following to the lxc config:
```
mp0: /zfs1/downloads,mp=/media/downloads
mp1: /zfs1/tvshows,mp=/media/tvshows
lxc.idmap: u 0 100000 65535
lxc.idmap: g 0 100000 1000
lxc.idmap: g 1000 1000 1
lxc.idmap: g 1001 101001 64534
```
Then you only need to give the `media` group access to the folders

**Reply by deku-m** [reply]
> For unpriviliged containers you can use idmapping in the config. I have set up my *arr lxcs to all use the `media` group (1000), and added the following to the lxc config:
> 
> ```
> mp0: /zfs1/downloads,mp=/media/downloads
> mp1: /zfs1/tvshows,mp=/media/tvshows
> lxc.idmap: u 0 100000 65535
> lxc.idmap: g 0 100000 1000
> lxc.idmap: g 1000 1000 1
> lxc.idmap: g 1001 101001 64534
> ```
> 
> Then you only need to give the `media` group access to the folders

The media group is created on proxmox then or inside the LXC?

**Reply by lordalisat** [reply]
Both. I believe they can have different GIDs ass well, just make sure to set it up correctly in the idmap.
It would be:
```
lxc.idmap: g 0 100000 $LXCGID
lxc.idmap: g $LXCGID $PROXMOXGID 1
lxc.idmap: g ${LXCGID+1} 10${$LXCGID + 1} 64534
```
